### PR TITLE
feat(sms): support server template resource

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
similar to #1805 

new resource: **huaweicloud_sms_server_template**
Manages an SMS server template resource within HuaweiCloud.

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/sms' TESTARGS='-run TestAccServerTemplate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sms -v -run TestAccServerTemplate -timeout 360m -parallel 4
=== RUN   TestAccServerTemplate_basic
=== PAUSE TestAccServerTemplate_basic
=== RUN   TestAccServerTemplate_existing
=== PAUSE TestAccServerTemplate_existing
=== CONT  TestAccServerTemplate_basic
=== CONT  TestAccServerTemplate_existing
--- PASS: TestAccServerTemplate_basic (57.21s)
--- PASS: TestAccServerTemplate_existing (103.59s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sms       103.679s
```

### CODE REVIEW

File: huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+			"vpc_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildVpcOpts(client *golangsdk.ServiceClient, vpcID string) (*templates.VpcRequest, error) {
+	if vpcID != "" && vpcID != AutoCreate {
```
Comment (hmbig2) [inline huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go:108]: pls describe `autoCreate` in the document

File: huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+				Name: AutoCreate,
+			},
+		}, nil
+	}
+
+	request := make([]templates.NicRequest, len(nics))
+	for i, subnetID := range nics {
+		if subnetID == AutoCreate {
+			request[i] = templates.NicRequest{
+				Id:   AutoCreate,
```
Comment (hmbig2) [inline huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go:141]: ditto

File: huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go:108]: done

File: huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/sms/resource_huaweicloud_sms_server_template.go:141]: done

**Pull request merged**